object/object.go

```diff
@@ -342,3 +344,39 @@ func (m Macro) Inspect() string {
 	out.WriteString("}")
 	return out.String()
 }
+
+type ExtFunction func(args []Object) Object
+
+type Extension struct {
+	Name     string
+	MinArgs  int
+	MaxArgs  int
+	ArgTypes []Type
+	Callback ExtFunction
+}
+
+func (e *Extension) Usage(out *strings.Builder) {
+	for i := 1; i <= e.MinArgs; i++ {
+		coma := ""
+		if i > 1 {
+			coma = ", "
+		}
+		t := strings.ToLower(e.ArgTypes[i-1].String())
+		out.WriteString(fmt.Sprintf("%s%s%d", coma, t, i))
+	}
```

Comment:
Can't all this be replaced by a strings.Join(whatever, ", ")

I mean the lowercase can be applied on everything no?

Reply:
type can be different

though maybe float1, int2, string3 is weirder, I sort of copied that code from fortio/cli arg help

Reply:
Ok

Reply:
maybe I should reset `i` each time type changes but then if it's

int, float, int -> int1, float1, int2

is a bit harder... maybe I drop the `i` altogether

Reply:
It's worth trying

Reply:
done